### ISSUE TITLE
feat:

### ISSUE THREAD

**Issue by RodrigoTR04** [andrewbenington/OpenHome#36]
Description
Allow for a toggle that will make save files read-only, some users will only use it to easily move Pokemon to other games while wanting to keep the original one on their save file, so allowing the save file to be read only will make sure they're not moving and then forgetting where Pokemon went.

**Comment by evanwporter** [comment]
Maybe I'm misunderstanding but you want to make a save read-only, thus allowing mons to be duplicated? If so PKHex already does this... Duplication would break the spirit of what OpenHome is.

**Comment by RodrigoTR04** [comment]
The point is not duplication, the point is to not alter the original save file, what the intent of this feature request is tied to https://github.com/andrewbenington/OpenHome/issues/34 where a user would be able to dump all their pokemon into OpenHome and then afterwards export it all in a single file for easy of carrying/storing.

**Comment by evanwporter** [comment]
Why does it need to be read only then?

**Comment by evanwporter** [comment]
>Because when you're importing dozens of thousands of Pokemon from multiple save files it would be ideal to have the option to not alter those files.

Is that not duplication then, to have one mon in a save and another in OH?

**Comment by RodrigoTR04** [comment]
It is duplication, but the point is not to use the copies, OpenHome is essentially a database, you want an entry there to then be able to locate where the original one is. Searching across dozens of save files is not fun.

**Comment by evanwporter** [comment]
I'm sorry but this probably won't be implemented. But if this is something you really want then I recommend tweaking the code yourself. It shouldn't be that hard either. You would need to look at the `GXSAV.ts` file `prepareBoxesAndGetModified` method and just comment out the line where it pushes to the `changedMon` array.

**Comment by RodrigoTR04** [comment]
The feature request is still vaild, even if it doesn't get implemented. I talked directly to @andrewbenington before opening a batch of feature requests and bug reports. And I do believe it would be a useful option. I'm not a programmer, and I don't plan on tweaking with this myself.

**Comment by andrewbenington** [comment]
@RodrigoTR04 your use case is definitely a valid one and probably a common one. I think the ability to search across many save files without moving mins out of those save files is a feature that should be added.

Ideally, though, that search feature would not require copying the moms into OpenHome boxes, because as @evanwporter pointed out, that goes against the spirit of OpenHome. So I think the best way to do this would be to use an interface like the Sort page to open save files in bulk and search through them without copying any mon data to the boxes. 

Allowing saves to be read-only is also a good idea independent of the search use case.

For now, though, this feature is going to take a backseat to the many basic features that are still broken or missing, like naming save boxes and a more robust sort/search page

**Comment by evanwporter** [comment]
Ah @RodrigoTR04 I think I misunderstood what you were asking for, you'd like a search feature where in the mon origin save/location is saved, that way you can lookup where a mon is using a reference mon from OH?

**Comment by RodrigoTR04** [comment]
Exactly, I'd like to be able to dump all the Pokemon I have across dozens of save files and then export it all in a single file afterwards. It's not practical with PKHeX because I have to dump it all into individual pokemon files into a folder and searching through that is extremely inconvenient, there's also the fact that if I want to transfer all those pokemon I need to keep zipping the folder where they're being stored. 

The fact that OpenHome offers "unlimited boxes" while not having to stick to a specific pokemon file type is a major plus. with PKHeX I can only view the pokemon details in the format of the save file loaded at the time, so if I want to be able to view a generation 3 pokemon caught  in USUM, but my loaded save file is from RSE PKHeX won't let me, I'll have to load an RSE save file before being able to view it, it's extremely inconvenient having to keep changing loaded save files or opening more instances of PKHeX.
